data/transactions/logic/eval.go

```diff
@@ -3522,10 +3548,52 @@ func (cx *EvalContext) stackIntoTxnField(sv stackValue, fs txnFieldSpec, txn *tr
 		txn.AssetReceiver, err = cx.availableAccount(sv)
 	case AssetCloseTo:
 		txn.AssetCloseTo, err = cx.availableAccount(sv)
-
-	// acfg likely next
-
-	// afrz seems easy but not high demand
+	// AssetConfig
+	case ConfigAsset:
+		txn.ConfigAsset, err = cx.availableAsset(sv)
+	case ConfigAssetTotal:
+		txn.AssetParams.Total, err = sv.uint()
+	case ConfigAssetDecimals:
+		var decimals uint64
+		decimals, err = sv.uint()
+		if err == nil {
+			if decimals > uint64(cx.Proto.MaxAssetDecimals) {
+				err = fmt.Errorf("too many decimals")
+			} else {
+				txn.AssetParams.Decimals = uint32(decimals)
+			}
+		}
+	case ConfigAssetDefaultFrozen:
+		txn.AssetParams.DefaultFrozen, err = sv.bool()
+	case ConfigAssetUnitName:
+		txn.AssetParams.UnitName, err = sv.string(cx.Proto.MaxAssetUnitNameBytes)
+	case ConfigAssetName:
+		txn.AssetParams.AssetName, err = sv.string(cx.Proto.MaxAssetNameBytes)
+	case ConfigAssetURL:
+		txn.AssetParams.URL, err = sv.string(cx.Proto.MaxAssetURLBytes)
+	case ConfigAssetMetadataHash:
+		if sv.Bytes == nil {
+			err = fmt.Errorf("ConfigAssetMetadataHash must be bytes")
+		} else if len(sv.Bytes) > 32 {
```

Comment:
I considered that, but I think I found that it's ok to submit, say, 6 bytes of metdatahash in a txn, and it gets copied in this way.  This is in libgoal:
```
	if len(metadataHash) > len(tx.AssetParams.MetadataHash) {
		return tx, fmt.Errorf("asset metadata hash %x too long (max %d bytes)", metadataHash, len(tx.AssetParams.MetadataHash))
	}
	copy(tx.AssetParams.MetadataHash[:], metadataHash)
```
So I wanted to maintain this interface.  Do SDKs enforce 32 bytes?

Reply:
Some of the SDKs (Javascript, Python, and maybe Java) enforce that the metadata hash must be 32 byets -- in my opinion this is the right thing to do and Goal and the Go SDK are in the wrong. The reason being it's very confusing if you set the metadata hash to `0x1234`, but when you check it later you actually get `0x12340000...`. This would be especially concerning in a smart contract.

Unless we actually want to make this field dynamically sized, I strongly suggest sticking with 32-byte inputs.

Reply:
That is a good point (that it comes back differently).  Would you say it's a bug in our transaction processing that we allow short metadatahashes? (if we do, I haven't 100% confirmed) it.  I think I would say it is, based on the same reasoning.

I'm willing to enforce 32 bytes here, I just hate being different from "native" processing.

Reply:
Actually, since this is libgoal, we might not allow it at the txn processing level, but libgoal is being a weirdo.  It is perhaps sending a 32 byte msgpack field, padded with 0s.  In *that* case, I'm 100% on board.

Reply:
> It is perhaps sending a 32 byte msgpack field, padded with 0s

Yes, that's exactly what's happening. The issue is not with txn processing, but rather with txn creation.

Reply:
Sold!